### PR TITLE
[Keymap] Added personal Kyria Keymap

### PR DESCRIPTION
**Pull request by john-ezra**
## Description

Added john-ezra Kyria keymap

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/splitkb/kyria/keymaps/john-ezra/config.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#pragma once
```
Comment (drashna) [inline keyboards/splitkb/kyria/keymaps/john-ezra/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/splitkb/kyria/keymaps/john-ezra/config.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+#ifdef OLED_DRIVER_ENABLE
+  #define OLED_DISPLAY_128X64
+#endif
+
+#ifdef RGBLIGHT_ENABLE
+  #define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/splitkb/kyria/keymaps/john-ezra/config.h:8]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/splitkb/kyria/keymaps/john-ezra/oled.c [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+					}
+				}
+			}
+			//then move the entire graph one pixel to the right
+			oled_pan(false);
+			//refresh the timer for the next iteration
+			timer = timer_read();
+		}
+		//format current WPM value into a printable string
+		sprintf(wpm_text,"%i", currwpm);
```
Comment (drashna) [inline keyboards/splitkb/kyria/keymaps/john-ezra/oled.c:96]: Just a heads up, sprintf is pretty expensive, in terms of resources. 

This uses 1.5k less:
```c
    char    wpm_counter[4];
    wpm_counter[3] = '\0';
    wpm_counter[2] = '0' + n % 10;
    wpm_counter[1] = (n /= 10) % 10 ? '0' + (n) % 10 : (n / 10) % 10 ? '0' : ' ';
    wpm_counter[0] = n / 10 ? '0' + n / 10 : ' ';
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/splitkb/kyria/keymaps/john-ezra/oled.c (hunk 3 above)
Comment (john-ezra) [reply keyboards/splitkb/kyria/keymaps/john-ezra/oled.c]: Done, got it working, thanks for the tip.

**Review by drashna — APPROVED** [review]

File: keyboards/splitkb/kyria/keymaps/john-ezra/oled.c [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            oled_write_P(PSTR("Undefined\n"), false);
+    }
+
+		render_bootmagic_status();
+
+// Host Keyboard LED Status
+    uint8_t led_usb_state = host_keyboard_leds();
+//    oled_write_P(IS_LED_ON(led_usb_state, USB_LED_NUM_LOCK) ? PSTR("NUMLCK ") : PSTR("       "), false);
+    oled_write_P(IS_LED_ON(led_usb_state, USB_LED_CAPS_LOCK) ? PSTR("CAPLCK ") : PSTR("       "), false);
+//    oled_write_P(IS_LED_ON(led_usb_state, USB_LED_SCROLL_LOCK) ? PSTR("SCRLCK ") : PSTR("       "), false);
```
Comment (fauxpark) [inline keyboards/splitkb/kyria/keymaps/john-ezra/oled.c:135]:
```suggestion
    led_t led_state = host_keyboard_led_state();
    oled_write_P(led_state.caps_lock ? PSTR("CAPLCK ") : PSTR("       "), false);
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/splitkb/kyria/keymaps/john-ezra/keymap.c [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+  QWERTY,
+  LOWER,
+  RAISE,
+  CPY_PST
+};
+
+#define BSP_CMD MT(MOD_LGUI, KC_BSPC)
+#define SFT_ENT MT(MOD_LSFT, KC_ENT)
+#define NKRO MAGIC_TOGGLE_NKRO
+#define OS_SWAP MAGIC_TOGGLE_CTL_GUI
```
Comment (drashna) [inline keyboards/splitkb/kyria/keymaps/john-ezra/keymap.c:38]: `NK_TOGG` and `CG_TOGG` do this as well.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**